### ISSUE TITLE
Field's condition filter Directus interface does not allow to input the correct filter

### ISSUE THREAD

**Issue by Pachat** [directus/directus#12753]
### Describe the Bug

Because of [dynamic-variables limitations concerning the use of $CURRENT_USER](https://docs.directus.io/configuration/filter-rules/#dynamic-variables) in conditions filters, we can however refer to `user_created` in the way described below.

The problem is that the Directus interface does not allow to input the correct filter and you have to Edit the raw.

A solution could be by documenting the way of doing things, 
or, better, by having the click on `user_created` not only opening the nested properties but also feeding the filter.
Event better, **a double click on `user_created` feeding the filter with `user_created` is a logical solution from the user POV**

Use case: 
- online student homework with a hidden field for the professor's exclusive use.

### To Reproduce

1. A collection with  `user_created`,  `input_field1` and `input_field2`
2. To have `input_field2` not accessible to the `user_created`, the filter that works and you want to build is
```
{
	"_or": [
		{
			"user_created": {
				"_eq": "$CURRENT_USER"
			}
		},
		{
			"user_created": {
				"_null": true
			}
		}
	]
}
```
and check `Hidden on Detail`

### Errors Shown

The Directus Interface does not allow to build such a filter: clicking on `User Created` only opens the nested properties. So you end with:
```
{
	"_or": [
		{
			"user_created": {
				"id": {
					"_eq": "$CURRENT_USER"
				}
			}
		},
		{
			"user_created": {
				"id": {
					"_null": true
				}
			}
		}
	]
}
```
which does not work.

### What version of Directus are you using?

9.7.1

### What version of Node.js are you using?

16.14.0

### What database are you using?

SQLite version 3.37.0

### What browser are you using?

Firefox and Edge

### How are you deploying Directus?

locally

**Comment by azrikahar** [comment]
> The Directus Interface does not allow to build such a filter: clicking on `User Created` only opens the nested properties.

This is the same issue as #12040, and the good news is it was fixed by #12320 in `v9.8.0`!

With that said, I do acknowledge that you may have held off from the upgrade due to #10842 (just an assumption based on your version & database). However, technically this particular filter interface issue has been fixed so I'll have to mark this as resolved. Hope that's understandable 👍

**Comment by Pachat** [comment]
Thanks for your reply.  Nice to hear that it has been fixed.
I hope to be able soon to follow the updates as soon as #10842 is fixed, 
or to have a path to manually migrate the database.
